src/aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -1062,7 +1062,7 @@ def presubmit(self, folder: Folder) -> CalcInfo:
 
         def encoder(obj):
             if dataclasses.is_dataclass(obj):
-                return dataclasses.asdict(obj)
+                return dataclasses.asdict(obj)  # type: ignore[arg-type]
```

Comment:
Not sure I should cast the type here, for the readability, I ignore the error.

Reply:
Would try to solve this in a subsequent PR